### PR TITLE
Shift suggestions to PlansView

### PR DESCRIPTION
**Pull request by tanruiquan**
Improve UI

### CODE REVIEW

File: TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift [hunk 1]
```diff
@@ -148,3 +153,35 @@ extension PlansViewModel {
...
+                            TransportationOptionsView(viewModel: viewModel)
+                        }
+
+                    SuggestionButton(title: "Conflict", subtitle: "There are clashing confirmed plans") {
+                        print("Implement conflict page")
+                    }
+                }
+            }
+        }
+        .padding()
```
Comment (keanecjy) [inline TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift]: Why is it in the view model though?

Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift]: If we are going to do conflict checking or nearby places recommendation then there'll be quite a bit of logic that's why i push it to the viewModel.

Comment (keanecjy) [reply TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift]: I don't think we should have any view elements in view models. We can either shift it to some view util/factory or just keep it in the view

File: TourMate/TourMate/Frontend/Plan/Views/TransportationOptionsView/TransportationOptionsViewModel.swift [hunk 2]
```diff
@@ -89,7 +90,7 @@ class TransportationOptionsViewModel: ObservableObject {
         }
     }
 
-    @ViewBuilder func makeTransportationOptionsCellView(_ result: RoutingResult) -> some View {
+    func makeTransportationOptionsCellView(_ result: RoutingResult) -> some View {
```
Comment (keanecjy) [inline TourMate/TourMate/Frontend/Plan/Views/TransportationOptionsView/TransportationOptionsViewModel.swift]: Could we shift out this part as well? VMs shouldnt be touching UI elements based on MVC style architecture, cuz it makes it hard for the views to be replaced.

File: TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift (hunk 1 above)
Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansViewModel.swift]: Alright i'll go and shift it

**Review by keanecjy — APPROVED** [review]

**Pull request merged**